### PR TITLE
Fix Sentry by adding integrations

### PR DESCRIPTION
**Pull request by tahb**
# Changes in this PR

I had hoped that configuring tracing would start to send transactions through to Sentry. It appears not, even on production where sampling is 1. I've not connected this locally to Sentry and can see transactions coming through!

Now we copy the rest of the Make a Recall set up[1] that adds integrations which, according to the commends, "enables HTTP call tracing".

We filter out requests to /assets to prevent spam and anonymise the username.

[1] https://github.com/ministryofjustice/manage-recalls-ui/blob/c3b4fd4486b26da47c086778fdc89e7a6befcde0/server/middleware/setUpSentry.ts

## Screenshots of UI changes

(Note the usernames that appear in earlier requests and then are filtered out in later requests after noticing them)

Screenshot 2023-02-24 at 12 18 38

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/middleware/setUpSentry.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
 import * as Sentry from '@sentry/node'
-import '@sentry/tracing'
+import * as Tracing from '@sentry/tracing'
 import express from 'express'
 import applicationVersion from '../applicationVersion'
 import config from '../config'
 
 export function setUpSentryRequestHandler(app: express.Express): void {
   if (config.sentry.dsn) {
+    const router = express.Router()
```
Comment (bryangaledxw) [inline server/middleware/setUpSentry.ts]: I think you'll need to call  `app.use(router)` for this to do anything. In make-recalls-ui, setupSentry returns the router, and `app.use(setUpSentry())` is called in app.ts

Comment (bryangaledxw) [reply server/middleware/setUpSentry.ts]: Also, I expect the call to `router.use(Sentry.Handlers.tracingHandler())` in make-recalls-ui is significant

Comment (tahb) [reply server/middleware/setUpSentry.ts]: Ah I didn't see this feedback before pushing my latest changes. I did end up removing this line 👍 

And yes that line was very significant as it turns out 😆

File: server/middleware/setUpSentry.ts [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+import * as Tracing from '@sentry/tracing'
 import express from 'express'
 import applicationVersion from '../applicationVersion'
 import config from '../config'
 
 export function setUpSentryRequestHandler(app: express.Express): void {
   if (config.sentry.dsn) {
+    // Prevent usernames which are PII from being sent to Sentry
+    // https://docs.sentry.io/platforms/python/guides/logging/data-management/sensitive-data/#examples
+    const anonymousID = Math.random().toString()
```
Comment (bryangaledxw) [inline server/middleware/setUpSentry.ts]: We're generally using `roomId`, `premisesId` etc., would change to `anonymousId` for consistency

**Review by bryangaledxw — APPROVED** [review]
Once incredibly picky point but otherwise looks good

File: server/middleware/setUpSentry.ts (hunk 2 above)
Comment (tahb) [reply server/middleware/setUpSentry.ts]: Didn't spot that, will change! 👍

**Pull request merged**